upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -489,7 +489,7 @@ func (v *AutoscalingGroup) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *Autos
 			if setup(request).LaunchTemplate == nil {
 				setup(request).LaunchTemplate = &autoscaling.LaunchTemplate{
 					LaunchTemplateSpecification: &autoscaling.LaunchTemplateSpecification{
-						LaunchTemplateId: changes.LaunchTemplate.ID,
+						LaunchTemplateId: e.LaunchTemplate.ID,
```

Comment:
> I don't know the reason for this logic. If anyone knows why, please tell me.

My mistake. Thanks for fixing.
Could you do same on line 457?

Reply:
In line 457 `changes.LaunchTemplate` is not nil because it is checked on line 455. So I don't think it needs to be changed.

Reply:
Just for consistency. I agree it won't improve anything.

Reply:
OK, I see. I will add it.

Reply:
I added: https://github.com/kubernetes/kops/pull/10742/commits/f22bafb43fb26388808c183fe6d062ba7510f4ca

Reply:
👍 
Can you also squash the 2 commits?

Reply:
I squashed and force pushed.